classes/form/guestlink_access_form.php

```diff
@@ -0,0 +1,104 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Config all BigBlueButtonBN instances in this course.
+ *
+ * @copyright Kevin Pham <kevinpham@catalyst-au.net>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+namespace mod_bigbluebuttonbn\form;
+
+defined('MOODLE_INTERNAL') || die();
+
+require_once($CFG->dirroot . '/mod/bigbluebuttonbn/locallib.php');
+require_once($CFG->libdir . '/formslib.php');
+require_once($CFG->dirroot . '/mod/bigbluebuttonbn/mod_form.php');
+
+/**
+ * Moodle class for mod_form.
+ *
+ * @copyright Kevin Pham <kevinpham@catalyst-au.net>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class guestlink_access_form extends \moodleform {
+
+    /**
+     * Define (add) particular settings this activity can have.
+     *
+     * @return void
+     */
+    public function definition() {
+        global $CFG, $DB, $OUTPUT, $PAGE;
+        $mform = &$this->_form;
+
+        // Guest Access URL.
+        $inviteparticipantselement = $mform->createElement('text', 'guestlinkurl', get_string('view_guestlink_label', 'bigbluebuttonbn'), ['readonly' => true, 'size' => 60]);
+        $mform->setType('guestlinkurl', PARAM_TEXT);
+        $mform->setDefault('guestlinkurl', $this->_customdata['guestlink']['url']);
+        $inviteparticipantsgroup = [];
```

Comment:
Picky but the readability would benefit from a linebreak between the inputs and the button group